api/src/main/java/org/apache/iceberg/types/PruneColumns.java

```diff
@@ -24,13 +24,27 @@
 import org.apache.iceberg.Schema;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.types.Types.ListType;
+import org.apache.iceberg.types.Types.MapType;
+import org.apache.iceberg.types.Types.StructType;
 
 class PruneColumns extends TypeUtil.SchemaVisitor<Type> {
   private final Set<Integer> selected;
+  private final boolean selectFullTypes;
 
-  PruneColumns(Set<Integer> selected) {
+  /**
+   * Visits a schema and returns only those element's whose id's have been passed as selected
```

Comment:
I think you may want to rephrase this because the part about "only those elements whose ids have been passed as selected" is a bit unclear to me. I would say "Visits a schema and returns only the fields selected by the ID set." Then in the description you could get more specific about `selectFullTypes`: "when true, selecting a nested type selects all subfields. otherwise, select produces empty structs." And you can also say that selecting maps and lists directly when `selectFullTypes` is enabled is ambiguous and therefore not allowed